core/loadpoint.go

```diff
@@ -1062,8 +1062,8 @@ func (lp *Loadpoint) fastCharging() error {
 	return err
 }
 
-// pvScalePhases switches phases if necessary and returns if switch occurred
-func (lp *Loadpoint) pvScalePhases(sitePower, minCurrent, maxCurrent float64) bool {
+// pvScalePhases switches phases if necessary and returns new target current if switch occurred
+func (lp *Loadpoint) pvScalePhases(sitePower, minCurrent, maxCurrent float64) float64 {
```

Comment:
I would use go's language feature to have mutliple return values instead, i.e. keep the boolean, and add the float as additional secondary parameter. That way we can get rid of the "magic value (-1)".

Reply:
Feel free to change it :-)

Reply:
Braucht es das überhaupt?
Entsprechend der Diskussion dazu könnte man auch erstmal mit `minCurrent` weitermachen.

Reply:
Ja - aber warum minCurrent nehmen, wenn man einen besseren Wert bekommen kann, der >= minCurrent ist?

Reply:
Kann man das?
In den "Ewigkeiten" der umschaltbedingten Ladepause kann viel passieren...

Reply:
> Kann man das?
>
Ja - denn mal angenommen man lädt mit 16A @ 1p und das Fahrzeug oder der Charger können nur 2p, dann kann man mit sagenhaften 8A weiter laden
> In den "Ewigkeiten" der umschaltbedingten Ladepause kann viel passieren...
>
Sehe ich genauso - deswegen kann man es auch gleich so lassen wie es ist und das Anliegen von [marioth1](https://github.com/marioth1) ignorieren. Oder man macht es richtig(er).

Reply:
@GrimmiMeloni: Changed it as suggested.